### PR TITLE
Support for API Keys based pack install and k/v setup, pack versioning, centos8 fixes, add validate_output_schema option, refactoring

### LINKED ISSUE #334: Add the ability to specify a pack version in the pack resource
Right now it looks like the pack resource is only validating the pack exists before the module installs the pack.  Allowing pack version specifications would ensure a particular version always exists on the Stackstorm server.

Thinking out loud here...
Add `version` to the [resource definition](https://github.com/StackStorm/puppet-st2/blob/master/lib/puppet/type/st2_pack.rb) - something like:
```
st2::pack { 'backups':
    version => "v2.0.0"
  }
```

The version is passed along to [this function](https://github.com/StackStorm/puppet-st2/blob/master/lib/puppet/provider/st2_pack/default.rb#L28) which [then executes](https://github.com/StackStorm/puppet-st2/blob/master/lib/puppet/provider/st2_pack/default.rb#L33) a modified command
```
output = exec_st2('pack', 'list', '-a', 'ref', 'version', '-j', '-t', token)
``` 

Then if that specific version isn't installed it gets installed [here](https://github.com/StackStorm/puppet-st2/blob/b744ee6130cb17aac0d06ecd1bab09a564e1ac37/tasks/pack_install.py#L15), passing along the version.

This is a massive oversimplification, no doubt.

### LINKED ISSUE #338: Add ability to install packs and set K/V using API Keys
Currently, the commands use the username/password provided to obtain a token for auth and then use the token in the requests. This can be replaced by an option to provide API keys in the config that can be then if present, can be directly used in the commands.

### PR DESCRIPTION
**Pull request by rush-skills**
This PR adds the support for installation of packs and K/V using API Keys as auth method, while also allowing to fix a version of the pack to be installed
It adds a flag to enable/disable validate_output_schema in st2.conf 
It also fixes a few bugs while running in CentOS (Stream) 8 and refactors some old style code.

Closes #338 
Closes #334

### CODE REVIEW

**Review by bishopbm1 — CHANGES_REQUESTED** [review]
This looks awesome. I would love to see something in the Readme explaining how an API key can be used vs the standard username password.

File: manifests/kv.pp [hunk 1]
```diff
@@ -14,12 +14,22 @@
...
 
+  if $apikey {
+    $_command = "st2 key set --api-key ${apikey} ${key} ${value}"
+    $_unless = "st2 key get --api-key ${apikey} ${key} | grep ${key}"
+  }
+  else {
+    $_command = "st2 key set ${key} ${value}"
+    $_unless = "st2 key get ${key} | grep ${key}"
+  }
+
```
Comment (nmaludy) [inline manifests/kv.pp:29]:
```suggestion
  if $apikey {
    $_cmd_flags = "--api-key ${apikey}"
  }
  else {
    $_cmd_flags = ''
  }
  $_command = "st2 key set ${_cmd_flags} ${key} ${value}"
  $_unless = "st2 key get ${_cmd_flags} ${key} | grep ${key}"
```

File: manifests/repo.pp [hunk 2]
```diff
@@ -17,11 +17,14 @@
 class st2::repo (
   Enum['present', 'absent'] $ensure = 'present',
   St2::Repository $repository = $st2::repository,
+  Boolean $manage_epel_repo = true
```
Comment (nmaludy) [inline manifests/repo.pp:20]:
```suggestion
  Boolean $manage_epel_repo = true,
```

missing trailing comma

File: test/integration/stackstorm/controls/services_test.rb [hunk 3]
```diff
@@ -27,14 +27,14 @@
   # st2auth
   describe port(9100) do
     it { should be_listening }
-    its('addresses') { should include '127.0.0.1' }
```
Comment (nmaludy) [inline test/integration/stackstorm/controls/services_test.rb:30]: These services should still listen on 127.0.0.1, not on 0.0.0.0.

File: manifests/profile/selinux.pp [hunk 4]
```diff
@@ -6,9 +6,18 @@
...
+        package { 'policycoreutils-python-utils':
+          ensure => present,
+        }
+      }
+    }
+    else {
+      if !defined(Package['policycoreutils-python']) {
+        package { 'policycoreutils-python':
+          ensure => present,
+        }
```
Comment (nmaludy) [inline manifests/profile/selinux.pp:20]:
```suggestion
    if (Numeric($facts['os']['release']['major']) >= 8) { # package was renamed in el8
      $package_policycoreutils = 'policycoreutils-python-utils'
    }
    else {
      $package_policycoreutils = 'policycoreutils-python'
    }
    if !defined(Package[$package_policycoreutils]) {
      package { $package_policycoreutils:
        ensure => present,
      }
    }
```

File: manifests/init.pp [hunk 5]
```diff
@@ -278,6 +278,7 @@
   $cli_silence_ssl_warnings = false,
   $cli_username             = $st2::params::admin_username,
   $cli_password             = $st2::params::admin_password,
+  $cli_apikey               = undef,
```
Comment (nmaludy) [inline manifests/init.pp:281]: these new parameters should have docs in the section above the class

**Review by nmaludy — CHANGES_REQUESTED** [review]

File: manifests/init.pp (hunk 5 above)
Comment (rush-skills) [reply manifests/init.pp:285]: Added the docstrings for the new params.

File: test/integration/stackstorm/controls/services_test.rb (hunk 3 above)
Comment (rush-skills) [reply test/integration/stackstorm/controls/services_test.rb]: Reverted the tests to check for 127.0.0.1, and changed services to bind to that IP as well

**Comment by rush-skills** [comment]
@nmaludy @bishopbm1 Thanks for the reviews. I have made the requested changes and added the docs for using API Keys in README.

**Comment by rush-skills** [comment]
@nmaludy @bishopbm1 @armab Can we get some eyes on this PR?

**Review by bishopbm1 — APPROVED** [review]
LGTM

File: .gitignore [hunk 6]
```diff
@@ -34,3 +34,4 @@
 /ci/
 Puppetfile.lock
 *.pyc
+.vscode/
```
Comment (nmaludy) [inline .gitignore:37]: FYI if you want this change to stick, need to add it to `.sync.yml` here: https://github.com/StackStorm/puppet-st2/blob/master/.sync.yml#L2

**Review by nmaludy — APPROVED** [review]

**Pull request merged**